opentracing-cdi-example/src/main/java/io/opentracing/contrib/cdi/example/InventoryNotificationService.java

```diff
@@ -0,0 +1,33 @@
+package io.opentracing.contrib.cdi.example;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.logging.Logger;
+
+/**
+ * This is the final call in the chain. This is an asynchronous stateless EJB, which obtains the span context
+ * via a method parameter. This bean is not intercepted in any way by us, so, the span context received is exactly
+ * the same as what was sent by the caller.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+public class InventoryNotificationService {
+    private static final Logger log = Logger.getLogger(InventoryNotificationService.class.getName());
+
+    @Inject
+    Tracer tracer;
+
+    @Inject
+    ActiveSpan parentSpan;
```

Comment:
Also, this service starts throwing exceptions when there is either no `tracer` or `parentSpan`.
Of the _tracer_ I can understand that an application developer chooses to always have one (with easy fallback to the noop tracer obviously).
For the _parent span_ I'm not so sure. You now have a service class that starts throwing exceptions on untraced call paths. For the example it may be okay, but maybe explain how you could make this work without an active span?